chemprop/data/splitting.py

```diff
@@ -25,8 +26,8 @@ class SplitType(EnumMapping):
     KMEANS = auto()
 
 
-def split_data(
-    datapoints: Sequence[MoleculeDatapoint],
+def make_split_indices(
```

Comment:
I felt that `split_data()` wasn't the right name for that function because it doesn't actually split the data. Instead it makes the indices to use to split the data. I think this is an important distinction to make because we only need two core functions in `chemprop.data.splitting`, (1) a function that uses a variety of methods to generate indices to split your data as you like and (2) a function that takes data and indices and does the actual splitting. I have implemented that with `make_split_indices()` (a rename of `split_data()`) and `split_data_by_indices()` which do what they say. Open to alternative names.